src/server/protocol_client.cc

```diff
@@ -56,17 +56,23 @@ static ProtocolClient::SSL_CTX* CreateSslClientCntx() {
   const auto& tls_key_file = GetFlag(FLAGS_tls_key_file);
   unsigned mask = SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
 
-  CHECK_EQ(1, SSL_CTX_use_PrivateKey_file(ctx, tls_key_file.c_str(), SSL_FILETYPE_PEM));
-  const auto& tls_cert_file = GetFlag(FLAGS_tls_cert_file);
+  // Load client certificate if given.
+  if (!tls_key_file.empty()) {
+    CHECK_EQ(1, SSL_CTX_use_PrivateKey_file(ctx, tls_key_file.c_str(), SSL_FILETYPE_PEM));
+    const auto& tls_cert_file = GetFlag(FLAGS_tls_cert_file);
```

Comment:
can `tls_cert_file` be empty in this context?

Reply:
if it is, the later statement will fail :)

Reply:
no, we do earlier validation in `ValidateClientTlsFlags`. will add a comment